### DISCUSSION TITLE
Action fails at Github API Calls when triggered but works after retry

### DISCUSSION THREAD

**Discussion by Ly0n** [community/community#26159] [Actions]
<p>Hi folks,</p>
<p>I’m accessing the Github API from an action.</p>
<p>This fails on triggers like push or release but if I retry the action by “re-run jobs” it work most of the time. </p>
<p>From the API I get back:</p>
<p>401 {“message”: “Bad credentials”, “documentation_url”: “<a href="https://developer.github.com/v3" rel="nofollow noopener">https://developer.github.com/v3</a>”}</p>
<p>API Calls are done by pygithub.</p>
<p>Is this a security feature?!</p>
<p>After testing multiple workarounds I feel a little furstrated :(.</p>
<p>Here the project:</p>
<p><a href="https://github.com/protontypes/seleryexample/runs/493813085" rel="nofollow noopener">https://github.com/protontypes/seleryexample/</a></p>

**Comment by Yanjingzhu** [comment]
<p>Do you mean the first successful workflow run failed before and then succeeded after re-run? </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/963f8512b54602ba11155bb4bb390a1a2cb4b4db.png" data-download-href="/uploads/short-url/lr9I5EflASUn1EhKroj7UCXl7kv.png?dl=1" title="re-run.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181095854-49f0bde1-0e9d-4aac-96dc-52262fee09fc.png" alt="re-run.png" data-base62-sha1="lr9I5EflASUn1EhKroj7UCXl7kv" width="690" height="269" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/963f8512b54602ba11155bb4bb390a1a2cb4b4db_2_690x269.png, https://aws1.discourse-cdn.com/github/original/2X/9/963f8512b54602ba11155bb4bb390a1a2cb4b4db.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/9/963f8512b54602ba11155bb4bb390a1a2cb4b4db.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/963f8512b54602ba11155bb4bb390a1a2cb4b4db_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">re-run.png</span><span class="informations">977×381 21.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Could you please <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/managing-a-workflow-run#enabling-step-debug-logging" rel="nofollow noopener">enable step debug logging</a> and share the debug logs for a failure one here? Do you call the Github API inside step " <strong>Run openselery on sources</strong>" ? Can you access to that api directly with curl or Postman? </p>

**Comment by Ly0n** [comment]
<blockquote>
<p><a class="mention" href="/yanjingzhu">@yanjingzhu</a> wrote:</p>
</blockquote>
<blockquote>
<p>Do you mean the first successful workflow run failed before and then succeeded after re-run? </p>
</blockquote>
<blockquote>
<p></p>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/963f8512b54602ba11155bb4bb390a1a2cb4b4db.png" data-download-href="/uploads/short-url/lr9I5EflASUn1EhKroj7UCXl7kv.png?dl=1" title="re-run.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/6413976/181095857-1dc6ad8a-8c31-4905-9e1b-27b5d6f5a03c.png" alt="re-run.png" data-base62-sha1="lr9I5EflASUn1EhKroj7UCXl7kv" width="690" height="269" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/963f8512b54602ba11155bb4bb390a1a2cb4b4db_2_690x269.png, https://aws1.discourse-cdn.com/github/original/2X/9/963f8512b54602ba11155bb4bb390a1a2cb4b4db.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/9/963f8512b54602ba11155bb4bb390a1a2cb4b4db.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/963f8512b54602ba11155bb4bb390a1a2cb4b4db_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">re-run.png</span><span class="informations">977×381 21.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
</blockquote>
<blockquote>
<p>Could you please <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/managing-a-workflow-run#enabling-step-debug-logging" rel="nofollow noopener">enable step debug logging</a> and share the debug logs for a failure one here? Do you call the Github API inside step " <strong>Run openselery on sources</strong>" ? Can you access to that api directly with curl or Postman? </p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<p>Yes, the first run fails in the step  <strong>Run openselery on sources </strong> and succeed after a retry. I didn’t know about the debug logging and will activate it. Use postman or curl could be a workaround, but I would appreciate to work with pygithub to make openselery more readable.  </p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>For enable step debug logging, you just need to add ACTIONS_STEP_DEBUG variable to repo&gt; Settings&gt; Secrets and set its value to true. Then you will see some purple lines in your logs. After you enable it, please share a failed workflow run debug logs here.  </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/a/ac247805acb39d9dd4f52f1224096462c82e88bb.png" data-download-href="/uploads/short-url/oyQg3KjndKhn3ukFlHVI3RvMU1d.png?dl=1" title="step debug.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181095860-22feb620-787c-4404-9846-a49880c4500b.png" alt="step debug.png" data-base62-sha1="oyQg3KjndKhn3ukFlHVI3RvMU1d" width="690" height="337" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/a/ac247805acb39d9dd4f52f1224096462c82e88bb_2_690x337.png, https://aws1.discourse-cdn.com/github/original/2X/a/ac247805acb39d9dd4f52f1224096462c82e88bb.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/a/ac247805acb39d9dd4f52f1224096462c82e88bb.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/a/ac247805acb39d9dd4f52f1224096462c82e88bb_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">step debug.png</span><span class="informations">1010×494 26.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>  </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/915e21a4ff72dc0e15b760711503d723865ccbfc.png" data-download-href="/uploads/short-url/kJYUlEXDixMhTSkeHu7RnP9tzwE.png?dl=1" title="debug log.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181095861-3e97dc33-1fac-4bf1-9a35-d7d9d7b17660.png" alt="debug log.png" data-base62-sha1="kJYUlEXDixMhTSkeHu7RnP9tzwE" width="550" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/915e21a4ff72dc0e15b760711503d723865ccbfc_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">debug log.png</span><span class="informations">638×579 11.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Ly0n** [comment]
<p>ACTIONS_RUNNER_DEBUG to true works. The behavior of the Github API calls looks stable now. Thanks <a class="mention" href="/yanjingzhu">@yanjingzhu</a> </p>
